### ISSUE TITLE
[HOLD for payment 2023-04-03] [$1000]  HTML characters are converted in chat (they shouldn't be converted)

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15508]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Enter `&lt;` or any other HTML character in the compose box
2. Click on send

## Expected Result
  ` &lt; `should remain the same (i.e. we don't want to convert HTML characters)

## Actual Result
   `&lt;` becomes the < sign

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.78-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/221594543-a616c80e-1b39-4c57-b9b7-2dfdc2f0f3d8.mov

https://user-images.githubusercontent.com/43996225/221594593-0ce68ce7-dc75-49b4-b19a-118429c75a92.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @esh-g 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1677316598380029

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012cc7a7ec930277c6</li>
        <li>Upwork Job ID: 1631456392901574656</li>
        <li>Last Price Increase: 2023-03-14</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @jliexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by jliexpensify** [comment]
I don't think this is a bug as typing &lt; should convert to <

Querying [here](https://expensify.slack.com/archives/C049HHMV9SM/p1677558613046939?thread_ts=1677316598.380029&cid=C049HHMV9SM).

**Comment by jliexpensify** [comment]
Asking our wider engineering team for a [second opinion](https://expensify.slack.com/archives/C03TQ48KC/p1677622840812139), thanks for your patience!

**Comment by aldo-expensify** [comment]
Somewhat related to: https://github.com/Expensify/App/issues/15131

**Comment by jliexpensify** [comment]
[Updated the Slack thread](https://expensify.slack.com/archives/C049HHMV9SM/p1677804251231419?thread_ts=1677316598.380029&cid=C049HHMV9SM), sending to Upworks.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~012cc7a7ec930277c6

**Comment by MelvinBot** [comment]
Current assignee @jliexpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @s77rt (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @NikkiWines (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by tienifr** [comment]
# Proposal

## Please re-state the problem that we are trying to solve in this issue.
If the message has HTML entities like &lt;, &nbsp, it's not showing as is in the chat.

## What is the root cause of that problem?

This is because we're unescaping HTML entities when displaying the message in the chat in here https://github.com/Expensify/App/blob/ce229bc1812ded576ce29d007fa3c03be8d1cd75/src/pages/home/report/ReportActionItemFragment.js#L123. We should not do this because the message should be shown as is.

## What changes do you think we should make in order to solve the problem?

We should remove `Str.htmlDecode` in https://github.com/Expensify/App/blob/ce229bc1812ded576ce29d007fa3c03be8d1cd75/src/pages/home/report/ReportActionItemFragment.js#L123
and other places if needed (basically any places that display the message should not be using `Str.htmlDecode`)

## What alternative solutions did you explore? (Optional)
NA

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Text with HTML entity is decoded on the chat item.

### What is the root cause of that problem?
We are decoding the text here:

https://github.com/Expensify/App/blob/2241edada07a73a6771a6fe5d74e294082222bd2/src/pages/home/report/ReportActionItemFragment.js#L121-L127

Why are we decoding it?

It is because when we send a chat, the chat we entered will be safely escaped/encoded. For example, if we send `& &amp;`, it will become `&amp; &amp;` and will be decoded later into `& &`. As you can see, the final text is different from what we enter before.

`& &amp -> & &` ❌ 

Here is the expected and actual escaped comment:

Expected: `&amp; &amp;amp;`
Actual: `&amp; &amp;`

### What changes do you think we should make in order to solve the problem?


We have 2 options here:
1. Replace the safe escape with just an escape.
https://github.com/Expensify/expensify-common/blob/fff304449214480065bf550e562ad39e681f6f95/lib/ExpensiMark.js#L326-L327

I'm not totally sure about this first option because expensimark may be used on another project as well and I have no context on why it is safely escaped.

2. Escape the text we want to add before passing it to the parser here.
https://github.com/Expensify/App/blob/9953eb896f60b676efd27541564a66326cbbcd34/src/libs/ReportUtils.js#L842-L845

**Comment by da7a90-backup** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Messages containing raw HTML entities/characters are formatted after sending, they should be displayed as is.
### What is the root cause of that problem?
Here `Str.htmlDecode(text)` decodes HTML entities within `props.fragment.text`
https://github.com/Expensify/App/blob/ce229bc1812ded576ce29d007fa3c03be8d1cd75/src/pages/home/report/ReportActionItemFragment.js#L123

This is why they are rendered. 
### What changes do you think we should make in order to solve the problem?
Either:
-remove `Str.htmlDecode(text)` and replace with just `text` 
-if it's still needed either add a `isText` parameter to the definition of the function with default `false` and return text as is if provided `true`
-change the level or scope of decode here https://github.com/Expensify/expensify-common/blob/main/lib/str.js#L92 

### What alternative solutions did you explore? (Optional)
I explored encoding the message again like another proposal suggests but that would be redundant and doesn’t address the root issue

**Comment by MelvinBot** [comment]
📣 @da7a90-backup! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by da7a90-backup** [comment]
Contributor details
Your Expensify account email: sidbarrack@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~018a8adf30b1e9aea5

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by s77rt** [comment]
@tienifr Thanks for the proposal! I think the RCA is not correct.

**Comment by s77rt** [comment]
@bernhardoj Thanks for the proposal! Same thing ^ the RCA is not correct.

**Comment by s77rt** [comment]
@da7a90-backup Thanks for the proposal. Same same ^^ the RCA is not correct.

**Comment by tgolen** [comment]
It's been a while, and I think that code was primarily used in OldDot. Looking at this bug report though, I think it's a feature and not a bug. If someone is copy/pasting a bunch of encoded text, I think we would want to convert it.

**Comment by esh-g** [comment]
@tgolen, But what if someone is sharing some HTML page code with these encodings? We would not want to convert in that case... And moreover, every platform like whatsapp and slack also don't convert

**Comment by tgolen** [comment]
> But what if someone is sharing some HTML page code with these encodings?

Is there something wrong with that?

**Comment by esh-g** [comment]
Well, It is better for consistency because the browser should always get `&lt;` instead of `<` and we do skip escaping characters like `&nbsp;` so I see no reason why `&lt;` should be an exception

**Comment by yuwenmemon** [comment]
Fun&nbsp;fact:&nbsp;I&apos;ve&nbsp;separated&nbsp;each&nbsp;word&nbsp;of&nbsp;this&nbsp;sentence&nbsp;with&nbsp;an&nbsp;`&nbsp;`

While I understand the viewpoint that what you type into the input is what should be displayed back to you. It's certainly not a paradigm that's universal across the internet. 

I do that it shouldn't be escaped within a code snippet, which is currently happening:
<img width="203" alt="Screenshot 2023-03-03 at 11 52 28 AM" src="https://user-images.githubusercontent.com/4741899/222813832-73db0a4f-7e44-43b4-b561-78cd0313ba75.png">

**Comment by yuwenmemon** [comment]
Some context on `safeEscape`...

I believe this was due to inconsistencies between Expensify and certain 3rd party integrations that were sending us categories with HTML entities already escaped.

So a category like `Yuwen & Tim` might be `Yuwen & Tim` from one integration and `Yuwen &amp; Tim` in another (i.e. already escaped). We did want to store all categories with HTML entities escaped. However if we just used escape the latter category would end up _displaying_ as `Yuwen &amp; Tim` (encoded as `Yuwen &amp;amp; Tim`)

**Comment by tienifr** [comment]
> I believe this was due to inconsistencies between Expensify and certain 3rd party integrations that were sending us categories with HTML entities already escaped.

Thanks for the context @yuwenmemon , I think if this is the case we should:
- Decide to escape or not at the integration level, for example if an integration sends already-escaped HTML entities then we'll not escape, if an integration sends not-escaped HTML entities then we'll escape. We shouldn't just use a general `safeEscape` to handle all those integrations because it can cause side effect like this issue.
- Then we should use only normal `escape` in https://github.com/Expensify/expensify-common/blob/2e6d46d1c6227974f27475aa00d8cd7963bdde8b/lib/ExpensiMark.js#L308 if the user types text in the composer

Considering that other major chat apps like Slack, WhatsApp show the typed text as is, I think we should do the same as well.

**Comment by s77rt** [comment]
Not overdue. Still discussing if we want to fix this...

**Comment by NikkiWines** [comment]
Re-assigning to @yuwenmemon (discussed 1:1)  as he has more context here

**Comment by s77rt** [comment]
@yuwenmemon Any updates on how to proceed here?

**Comment by pecanoro** [comment]
Let's put a HOLD on this issue until this PR https://github.com/Expensify/App/pull/15340 gets merged.

**Comment by pecanoro** [comment]
Actually, this one seems unrelated as it's encoding all messages not only IOU.

**Comment by yuwenmemon** [comment]
Right @pecanoro but we should probably mirror the behavior of the input in that PR. What do you think?

**Comment by pecanoro** [comment]
Yes, I agree!

**Comment by s77rt** [comment]
@yuwenmemon https://github.com/Expensify/App/issues/15508#issuecomment-1453267179

**Comment by yuwenmemon** [comment]
@s77rt based on the fact that we want to mirror the other inputs, double-escaping makes sense here.

**Comment by s77rt** [comment]
@yuwenmemon in that case, should we close the issue?

**Comment by yuwenmemon** [comment]
@s77rt Sorry to clarify I mean that HTML characters should **not** be converted in chat. Since if we look at the [other PR](https://github.com/Expensify/App/pull/15340), we're not going to escape them in names, IOUs, etc.

**Comment by s77rt** [comment]
@yuwenmemon Alright, so what's holding us from modifying [safeEscape](https://github.com/Expensify/expensify-common/blob/2e6d46d1c6227974f27475aa00d8cd7963bdde8b/lib/str.js#L106-L114) to `return _.escape(s);`

**Comment by MelvinBot** [comment]
@yuwenmemon @s77rt @jliexpensify this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by yuwenmemon** [comment]
Sorry, @s77rt - nothing is holding us! Let's pick a proposal. From the looks of it I'd say @bernhardoj is describing what we'd want to do here - do you agree?

**Comment by s77rt** [comment]
@yuwenmemon Can you revert the price back to $1000? I think none of the proposals got the right RCA but @bernhardoj proposed the right solution. Is that a blocker? if not let's hire him!

**Comment by yuwenmemon** [comment]
yeah the RCA thing was a bug - and agreed to revert to $1000. Thanks!

**Comment by MelvinBot** [comment]
📣 @bernhardoj You have been assigned to this job by @yuwenmemon!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by bernhardoj** [comment]
PR is ready for review.

cc: @s77rt

**Comment by s77rt** [comment]
@bernhardoj I think you implemented the solution in the wrong place, I'm expecting an upstream fix on `expensify-common`. With your current solution we have something like `_.escape(_.unescape(_.escape(s)))` which is a redundant, it should be simply `_.escape(s)`.

**Comment by s77rt** [comment]
@bernhardoj I see we are a bit lost in the terms here :sweat_smile: 
1. Allow double escaping: by calling `_.escape` twice
2. Allow double escaping: by not preventing it in the first place i.e. do not call `_.unescape` 

Both approaches will fix the issue, I just think that 2 is more straightforward cc @yuwenmemon thoughts here? Option 1 requires changes on E/App only while Option 2 requires an upstream fix (expensify-common).

**Comment by yuwenmemon** [comment]
Yes, @bernhardoj we're looking for you to do what you said [here](https://github.com/Expensify/App/issues/15508#issuecomment-1453054412):

> Replace the safe escape with just an escape.
> https://github.com/Expensify/expensify-common/blob/fff304449214480065bf550e562ad39e681f6f95/lib/ExpensiMark.js#L326-L327
> I'm not totally sure about this first option because expensimark may be used on another project as well and I have no context on why it is safely escaped.

**Comment by jliexpensify** [comment]
Hired @s77rt . @esh-g and @bernhardoj - I can't find you on Upworks, can you:

1. Send me your Upworks names or profiles
2. Edit your GH accounts to show your full names (so we can find you in the future)

**Comment by bernhardoj** [comment]
@yuwenmemon I see. But actually, I just realized that we may still need to escape it on E/App in case the text length is more than the max markup length.

https://github.com/Expensify/App/blob/9953eb896f60b676efd27541564a66326cbbcd34/src/libs/ReportUtils.js#L842-L845

**Comment by s77rt** [comment]
@bernhardoj Why would that be a problem for long text? The message will be sent directly to the backend as it is which is basically the expected behavior.

**Comment by bernhardoj** [comment]
@s77rt Yes, but the message is not escaped. So if we have html entities inside the long text, &amp; will become & again.

**Comment by s77rt** [comment]
Ah yes but I think that goes for everything else, the behavior will be different as on long text we are not sending html but plaintext e.g. send a long text that contain `*bold*` the text won't be in bold. The only reason for that difference seems due to performance.

**Comment by esh-g** [comment]
@jliexpensify 
Here is my Upwork profile: https://www.upwork.com/freelancers/~0186b1d7cc69656f22

I'll update my name to match this

**Comment by bernhardoj** [comment]
Yes, it won't be converted to HTML for long text, but we still need to escape it.

Assume this is a long text: `long text with *markdown* &amp; HTML entities` 

If we don't escape it, the text we will see become `long text with *markdown* & HTML entities` .

Notice the `&amp;` become `&` because we decode it inside `ReportActionItemFragment`.

**Comment by s77rt** [comment]
So the plan is:
For short texts: parse the message (markdown) and encode the text
For long texts: we can't parse it due to performance reasons but just encode it (assuming that  encoding is not a heavy process)

**Comment by bernhardoj** [comment]
Yes, correct. We won't change the parsing part, just the encoding/escaping. So, should we reopen the PR?

**Comment by s77rt** [comment]
We still need to fix this upstream. The `replace` method (in expensify-common) should not prevent double-escape. And for E/App for texts that are too long we just encode them if that's not a slow process.

**Comment by bernhardoj** [comment]
Should we just remove the `safeEscape` from `replace` method? So,
1. Apply the changes from closed PR
2. Remove `safeEscape` in expensify-common

**Comment by s77rt** [comment]
1. expensify-common (`replace` method):
```diff
-    let replacedText = shouldEscapeText ? Str.safeEscape(text) : text;
+    let replacedText = shouldEscapeText ? _.escape(text) : text;
```

2. E/App
```diff
-    return text.length < CONST.MAX_MARKUP_LENGTH ? parser.replace(text) : text;
+    return text.length < CONST.MAX_MARKUP_LENGTH ? parser.replace(text) : _.escape(text);
```

I'm expecting something like that ^

**Comment by bernhardoj** [comment]
Got it 👍. I will open the PR in a few hours.

**Comment by bernhardoj** [comment]
PR is up.

**Comment by sobitneupane** [comment]
@s77rt We are expecting this issue to solve [these issues](https://github.com/Expensify/App/pull/16055#issuecomment-1477525578) as well.

**Comment by bernhardoj** [comment]
@sobitneupane I don't know how to test the notification, but the LHN part, I have tested and the upcoming PR for this issue won't fix it. There is an issue reported for the LHN issue too here https://github.com/Expensify/App/issues/15920. You can take a look at the video and notice the LHN preview able to shows `&nbsp;` for a while, then converted to empty space. I'm guessing there is 2 decoding happen, once from the BE and once from the front-end. So, when we send `&nbsp;`,

1. Encoded to `&amp;nbsp;`
2. Decoded at backend becomes `&nbsp;`
3. Decoded at frontend becomes empty space

I take a look at the PR you linked and looks like the solution to it is to remove the decoding from FE. So, I think we should decide where we should do the decoding.

**Comment by s77rt** [comment]
@sobitneupane I'm not sure how those linked issues are related to what we are fixing here.
- What we are fixing here is related to what we send to the server (if user writes `&amp;` we will send `&amp;amp;` so the returned message gets decoded correctly to reflect what the user initially sent).
- The linked PR have no relation to what to read from the server (it does not alter when to decode or not to decode).

**Comment by pecanoro** [comment]
The text fragment from the back-end is already decoded, so we send `&nbsp;` and it should not change. There is definitely some extra decode somewhere that is decoding it to a blank space. This issue mentions HTML characters are converted in chat, so I would think we need to fix it here?

**Comment by s77rt** [comment]
@pecanoro This issue here is about what we send to the server and not about what receive from the server. So I'm not sure what you are referring to can be fixed here. Can you please create a new issue and assign me there? I'd like to verify the performed actions and actual/expected results of the new issue to better confirm if this can handled here or in a separated issue.

**Comment by s77rt** [comment]
I think decoding the text is what we are doing wrong here, but I'm expecting that to be fixed in this PR https://github.com/Expensify/App/pull/15340 Either way let's wait to get this issue fixed then we can see how to move forward.

**Comment by bernhardoj** [comment]
Expensify common PR is merged.

I have opened the App PR.

cc: @s77rt

**Comment by yuwenmemon** [comment]
Oops! I gotta create the Web-Expensify PR don't I?

**Comment by s77rt** [comment]
@yuwenmemon No, I don't think so.

**Comment by yuwenmemon** [comment]
@s77rt I do, because we use ExpensiMark in that code as well. Sorry I wasn't really asking I was saying I forgot to do it! (All taken care of now though!)

**Comment by s77rt** [comment]
Ah yes, I think this is only for Concierge?

**Comment by yuwenmemon** [comment]
Correctomundo!

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.89-0 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/16380


If no regressions arise, payment will be issued on 2023-04-03. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@s77rt / @yuwenmemon] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@s77rt / @yuwenmemon] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@s77rt / @yuwenmemon] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@jliexpensify] Determine if we should create a regression test for this bug.
- [ ] [@s77rt] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@jliexpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: The bug is upstream. But I think the upstream PR is https://github.com/Expensify/expensify-common/pull/232
- The offending PR has been commented on: Not sure this applies here. cc @yuwenmemon 
- A discussion in #expensify-bugs has been started: Same ^

___

**Regression Test Proposal**

1. Open any chat
2. Send `&amp;` as a message
3. Verify the sent text shows `&amp;`

**Comment by pecanoro** [comment]
I was going to close this [issue](https://github.com/Expensify/App/issues/15026) since the problem was almost the same as here and got fixed already, but I am wondering if that one deserves the reporting bonus instead of this one since the other one was created before.

**Comment by jliexpensify** [comment]
I personally think both issues deserve the reporting bonus - the contributors did both report valid bugs (even if they're similar), and the reporting bonus isn't large (only $250). We can also bring this up in #bug-zero if you wanted a second opinion?

**Comment by jliexpensify** [comment]
Looks like there are no regressions, so creating the GH for regression test and paying everyone.

**Comment by jliexpensify** [comment]
All paid, and job closed!
